platform/src/main/java/org/stellar/anchor/platform/SepConfig.java

```diff
@@ -158,22 +156,20 @@ CirclePaymentService circlePaymentService(
     return new CirclePaymentService(circlePaymentConfig, circleConfig, horizon);
   }
 
-  @Bean
-  Sep31DepositInfoGenerator sep31DepositInfoGeneratorCircle(
-      CirclePaymentService circlePaymentService) {
-    return new Sep31DepositInfoGeneratorCircle(circlePaymentService);
-  }
-
   @Bean
   Sep31DepositInfoGenerator sep31DepositInfoGenerator(
-      Sep31Config sep31Config, Sep31DepositInfoGenerator sep31DepositInfoGeneratorCircle) {
+      Sep31Config sep31Config,
+      CirclePaymentService circlePaymentService,
+      UniqueAddressIntegration uniqueAddressIntegration) {
     switch (sep31Config.getDepositInfoGeneratorType()) {
       case SELF:
         return new Sep31DepositInfoGeneratorSelf();
 
       case CIRCLE:
-        return sep31DepositInfoGeneratorCircle;
+        return new Sep31DepositInfoGeneratorCircle(circlePaymentService);
 
+      case API:
+        return new Sep31DepositInfoGeneratorApi(uniqueAddressIntegration);
```

Comment:
🙏  Now you're adding this, could you plz update the `PropertySep31Config` validation? @erika-sdf did a very good job there and we should make sure the validation gets up to date with new configurations.

Reply:
It'd be great to validate the unique_address endpoint there, in case `API` was configured as the deposit info generator.

Reply:
Added.